### DISCUSSION TITLE
sleep watch does not work as expected if run via brew service

### DISCUSSION THREAD

**Discussion by sajimon** [Homebrew/discussions#4844] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: e57a87cfe9c15f192cd4fc6695a2b3b7acf0ddf7
Last commit: 6 days ago
Core tap JSON: 10 Oct 18:24 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.0
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

When I run sleep watch using brew service functionality, it start as expected, but does not react to sleep/wake events. When I run it using terminal it works just fine (until I close terminal ofc.)

Terminal invokation 
`/usr/local/opt/sleepwatcher/sbin/sleepwatcher -V -s $HOME/.sleep -w $HOME/.wakeup&`

**Comment by isaacwisdom** [comment]
I am also having this same problem. Sleepwatcher runs scripts as expected when running from the terminal, but does not work when run using brew services.

**Comment by creechy** [comment]
+1 - on Sequoia, `sleepwatcher` works running it by hand in a terminal but not in a service for me, on Sequoia.

**Reply by creechy** [reply]
I figured out the problem, it was in my script not finding a command after the upgrade. It would be helpful if there were some way for brew services to capture output and log it somewhere. I ended up putting some redirect statements in my script to do this, for example
```
exec >> "/tmp/.sleepwatcher-${USER}.out" 2>&1
```

**Reply by SMillerDev** [reply]
There is, it's documented in https://docs.brew.sh/Formula-Cookbook#service-block-methods

**Reply by creechy** [reply]
Neat! Not familiar with the innards of Homebrew but I take it, just need to update the formulae to include the logging syntax.
